doc/admin-guide/plugins/icap.en.rst

```diff
@@ -0,0 +1,122 @@
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+
+
+The ICAP plugin enables passing along the response header and body to
+an external server for additional processing (generally scanning) using
+the ICAP protocol.  The scanner, depending on whether the file contains
+malicious content, will return a transformed version of the file. If the
+file is clean, then the exact same file will be returned. Otherwise, the
+scanner will block the file and return a message containing information
+regarding the infection.
+
+Installation
+------------
+
+This plugin is only built if the configure option ::
+
+    --enable-experimental-plugins
+
+is given at build time.
+
+To use this plugin, you need to compile the plugin and add the following line in plugin.config (assuming the compiled plugin library is called icap_plugin.so):
+
+.. code-block:: text
+
+   icap_plugin.so scanner_server_ip scanner_server_port avoid_port debug_enable
+
+There are 4 parameters to this plugin: the scanner_server_ip and scanner_server_port (standard ICAP port is 1344) is required to initiate connection to the ICAP server, and the avoid_port is specified to avoid processing traffic that appears on particular ports (e.g, requests passed via the parent select and already checked).  Parameter debug_enable will enable debug mode, which, in cases of ICAP server connection error or ICAP server too busy to handle the request, will return the origin response body to the client instead of an error message.
+
+ICAP
+----
+
+`ICAP <https://tools.ietf.org/html/rfc3507>`__ is a light-weight protocol for executing a remote procedure call on HTTP messages. It is very handy to request transformations on HTTP requests and responses since ICAP servers are set up on the edge of the Internet. Therefore, ICAP has become an industry standard for Anti Virus service.
+
+Virus Scanning via ICAP
+-----------------------
+
+Perform virus scanning via ICAP has several advantages over the traditional virus scanner deployment, which is to deploy the scanner locally on the same machine, is that since virus scanner is deployed on a remote server, it doesn't take up resources locally, so that the main thread on the server will be able to gain performance. And the fact that ICAP servers are also deployed on the edge, they can be physically close to the proxy, so there is very little latency on file transmission.
```

Comment:
May I suggest:
Performing virus scanning via ICAP has several advantages over the traditional local host virus scanner deployment model. By performing virus scanner on a remote server, local resources are not used, allowing the main thread on the server be more performant. ICAP servers are frequently deployed on the edge near |TS| proxies, adding minimal latency on file transmission.